tests/kernel/fp_sharing/float_disable/src/k_float_disable.c

```diff
@@ -49,13 +49,17 @@ static void usr_fp_thread_entry_2(void)
 
 void test_k_float_disable_common(void)
 {
+	ret = TC_PASS;
+
+	k_thread_priority_set(k_current_get(), K_PRIO_COOP(PRIORITY));
+
 	/* Create an FP-capable User thread */
 	k_thread_create(&usr_fp_thread, usr_fp_thread_stack, STACKSIZE,
 		(k_thread_entry_t)usr_fp_thread_entry_1, NULL, NULL, NULL,
 		K_PRIO_COOP(PRIORITY), K_USER | K_FP_OPTS,
 		K_NO_WAIT);
 
-	k_sleep(1);
+	k_yield();
```

Comment:
So... strictly I don't know if we document the exactly behavior of k_yield() with respect to threads of the same priority.  I mean, the implementation is that it goes back into the run queue, where it lives behind any threads that are already there, so this will absolutely work with the current scheduler (and it would be a little insane to have a scheduler that didn't behave like this).

If you want to be sure that usr_fp_thread_entry_1() runs here, it's probably cleaner to construct a higher priority for the new thread (e.g. subtract one from the current thread priority and assert that it's a valid priority)

Reply:
>So... strictly I don't know if we document the exactly behavior of k_yield() with respect to threads of the same priority

Thanks for looking into the details, @andyross  :) So I had a check in the documentation, and it seems it is aligned with the implementation, as well as with "sanity" :) Here is the text:
>Calling k_yield() puts the thread at the back of the scheduler’s prioritized list of ready threads, and then invokes the scheduler. All ready threads whose priority is higher or equal to that of the yielding thread are then allowed to execute before the yielding thread is rescheduled. If no such ready threads exist, the scheduler immediately reschedules the yielding thread without context switching.

So, I think we are good as long as we don't change the kernel scheduler API :)
I will, anyway, add some inline help text, to indicate what I expect from k_yield() in the test.

Reply:
>If you want to be sure that usr_fp_thread_entry_1() runs here, it's probably cleaner to construct a higher priority for the new thread (e.g. subtract one from the current thread priority and assert that it's a valid priority)

I don't want to overcomplicate the test; I prefer to make it easier to follow (for the reader that just wants to know how the logic behind the FP services)